File: src/api/MoRS.ManagementSystem.Application/DTOs/Email/CreateEmailRequest.cs
using System.ComponentModel.DataAnnotations;

namespace MoRS.ManagementSystem.Application.DTOs.Email;

public class CreateEmailRequest
{
    [Required]
    [StringLength(100, MinimumLength = 5)]
    public required string Subject { get; set; }

    [Required]
    [StringLength(2000, MinimumLength = 10)]
    public required string Body { get; set; }

    [Required]
    [MinLength(1)]
    public List<int> UserIds { get; set; } = null!;
}

File: src/desktop/lib/models/email/create_email_request.dart
class CreateEmailRequest {
  final String subject;
  final String body;
  final List<int> userIds;

  CreateEmailRequest({
    required this.subject,
    required this.body,
    required this.userIds,
  });

  factory CreateEmailRequest.fromJson(Map<String, dynamic> json) {
    return CreateEmailRequest(
      subject: json['subject'] as String,
      body: json['body'] as String,
      userIds: List<int>.from(json['userIds'] as List),
    );
  }

  Map<String, dynamic> toJson() {
    return {'subject': subject, 'body': body, 'userIds': userIds};
  }

  bool get isValid {
    return subject.length >= 5 &&
        subject.length <= 100 &&
        body.length >= 10 &&
        body.length <= 2000 &&
        userIds.isNotEmpty;
  }
}
